metrics/otel_metrics.go

```diff
@@ -263,59 +225,101 @@ func (o *OTelMetrics) Increment(name string) {
 	o.lock.Lock()
 	defer o.lock.Unlock()
 
-	if ctr, ok := o.counters[name]; ok {
-		ctr.Add(context.Background(), 1)
-		o.values[name]++
+	var err error
+	ctr, ok := o.counters[name]
+	if !ok {
+		ctr, err = o.initCounter(Metadata{
+			Name: name,
+		})
+
+		if err != nil {
```

Comment:
Given that we're ignoring the error anyway, why bother to have initCounter return an error?

Reply:
I'm returning it for the Register call so that we can know what the error is during registration